### PR TITLE
Filter profiling tool based on start time. 

### PR DESCRIPTION
**Pull request by parthosa**
Closes: #4323 

Added an option (`--start-app-time` and `-s`) in Profiling tool to filter event logs using application start time. This is similar to the one used in Qualification tools.

Examples:
1. `--start-app-time 1d <event-logs>`
```
22/06/27 11:38:02 INFO FilterAppInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514904001
22/06/27 11:38:02 INFO FilterAppInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514789396
22/06/27 11:38:03 INFO FilterAppInfo: Total number of events parsed: 6 for file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514789396
22/06/27 11:38:03 INFO FilterAppInfo: Total number of events parsed: 6 for file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514904001
22/06/27 11:38:03 WARN ProfileMain: No event logs to process after checking paths, exiting!
```
2. `--start-app-time 1m <event-logs>`
```
22/06/27 11:46:40 INFO FilterAppInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514789396
22/06/27 11:46:40 INFO FilterAppInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514904001
22/06/27 11:46:41 INFO FilterAppInfo: Total number of events parsed: 6 for file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514904001
22/06/27 11:46:41 INFO FilterAppInfo: Total number of events parsed: 6 for file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514789396
22/06/27 11:46:41 INFO Profiler: Threadpool size is 3
22/06/27 11:46:41 INFO ApplicationInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514904001
22/06/27 11:46:41 INFO ApplicationInfo: Parsing Event Log: file:/Users/psarthi/Desktop/Work/profiling/event-logs/local-1655514789396
```
Signed-off-by : Partho Sarthi [psarthi@nvidia.com](mailto:psarthi@nvidia.com)

### CODE REVIEW

**Comment by parthosa** [comment]
build

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AppFilterSuite.scala [hunk 1]
```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
```
Comment (nartal1) [inline tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AppFilterSuite.scala:2]: Nit: only 2022 should be mentioned as it's a new file.

**Review by nartal1 — COMMENTED** [review]
Please update the profiling tool doc - https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/docs/spark-profiling-tool.md#profiling-tool-options

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AppFilterSuite.scala (hunk 1 above)
Comment (parthosa) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AppFilterSuite.scala]: Resolved.

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala [hunk 2]
```diff
@@ -53,26 +56,50 @@ class AppFilterImpl(
...
+  private def filterEventLogsInternal(
+       apps: Seq[AppFilterReturnParameters],
+       appArgs: ProfileArgs): Seq[EventLogInfo] = {
+    val appTimeFiltered = if (appArgs.startAppTime.isSupplied) {
+      val msTimeToFilter = AppFilterImpl.parseAppTimePeriodArgs(appArgs.startAppTime)
+      apps.filter(_.appInfo.appStartInfo.exists(_.startTime >= msTimeToFilter))
+    } else {
+      apps
+    }
+
```
[the remark this answers is not in the document]
Comment (parthosa) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala]: Remove newline

**Comment by nartal1** [comment]
> Please update the profiling tool doc - https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/docs/spark-profiling-tool.md#profiling-tool-options

Could you please push the updated  `spark-profiling-tool.md`  file in this PR.

**Comment by nartal1** [comment]
build

**Review by nartal1 — APPROVED** [review]
LGTM.

**Comment by nartal1** [comment]
build

**Pull request merged**